src/sentry/tasks/assemble.py

```diff
@@ -530,71 +537,173 @@ def _create_artifact_bundle(
             # not support them, some files were not injected...).
             if (
                 organization is not None
-                and release
+                and self.release
                 and features.has(
                     "organizations:sourcemaps-bundle-indexing", organization, actor=None
                 )
             ):
-                # After we committed the transaction we want to try and run indexing.
-                _index_bundle_if_needed(
-                    org_id=org_id, release=release, dist=dist, date_snapshot=date_snapshot
+                # After we committed the transaction we want to try and run indexing by passing non-null release and
+                # dist. The dist here can be "" since it will be the equivalent of NULL for the db query.
+                self._index_bundle_if_needed(
+                    release=self.release,
+                    dist=(self.dist or NULL_STRING),
+                    date_snapshot=date_snapshot,
                 )
         else:
             raise AssembleArtifactsError(
                 "uploading a bundle without debug ids or release is prohibited"
             )
```

Comment:
I very much prefer early returns. then you can also un-indent all the rest.

Also, I thought we wanted to change this, so you can upload bundles without a release?

Reply:
You can, we check for debug ids in case you don't have a release set.